samples/java_layout/src/main/java/io/customer/android/sample/java_layout/ui/settings/SettingsActivity.java

```diff
@@ -140,16 +136,25 @@ private void setupObservers() {
         });
     }
 
-    private boolean isTrackingURLValid(String url) {
+    private boolean isHostURLInvalid(String url) {
         // Empty text is not considered valid
         if (TextUtils.isEmpty(url)) {
-            return false;
+            return true;
         }
 
-        Uri uri = Uri.parse(url);
-        String scheme = uri.getScheme();
-        // Since SDK does not allow tracking URL with empty host or incorrect schemes
-        return !TextUtils.isEmpty(uri.getAuthority()) && ("http".equals(scheme) || "https".equals(scheme)) && uri.getPath().endsWith("/");
+        try {
+            Uri uri = Uri.parse(url);
+            // Since SDK does not support custom schemes, we manually append http:// to the URL
+            // So the URL is considered invalid if it ends with a slash, contains a scheme, query or fragment
+            return url.endsWith("/")
+                    || !TextUtils.isEmpty(uri.getScheme())
+                    || !TextUtils.isEmpty(uri.getQuery())
+                    || !TextUtils.isEmpty(uri.getFragment());
```

Comment:
Same as https://github.com/customerio/customerio-android/pull/337#discussion_r1603970406

Reply:
Replied in linked comment.